pulse2percept/api.py

```diff
@@ -0,0 +1,423 @@
+        """Transforms an input stimulus to a percept
+
+        Parameters
+        ----------
+        stim : utils.TimeSeries|list|dict
+            There are several ways to specify an input stimulus:
+
+            - For a single-electrode array, pass a single pulse train; i.e.,
+              a single utils.TimeSeries object.
+            - For a multi-electrode array, pass a list of pulse trains; i.e.,
+              one pulse train per electrode.
+            - For a multi-electrode array, specify all electrodes that should
+              receive non-zero pulse trains by name.
+        t_percept : float, optional
+            The desired sampling time step (seconds) of the output. If None is
+            given, the output sampling time step will correspond to the input
+            time step (as defined by `stim` and `tm`).
+            Default: None.
+        tol : float, optional
+            Ignore pixels whose effective current is smaller than a fraction
+            `tol` of the max value.
+            Default: 0.05.
+
+        Returns
+        -------
+        A utils.TimeSeries object whose data container comprises the predicted
+        brightness over time at each retinal location (x, y), with the last
+        dimension of the container representing time (t).
+
+        Examples
+        --------
+        Stimulate a single-electrode array:
+
+        >>> import pulse2percept as p2p
+        >>> implant = p2p.implants.ElectrodeArray('subretinal', 0, 0, 0, 0)
+        >>> stim = p2p.stimuli.Psycho2Pulsetrain(tsample=5e-6, freq=50, amp=20)
+        >>> sim = p2p.Simulation("Single-electrode example", implant)
+        >>> percept = sim.pulse2percept(stim)  # doctest: +SKIP
+
+        Stimulate a single electrode ('C3') of an Argus I array centered on the
+        fovea:
+
+        >>> import pulse2percept as p2p
+        >>> implant = p2p.implants.ArgusI()
+        >>> stim = {'C3': p2p.stimuli.Psycho2Pulsetrain(tsample=5e-6, freq=50,
+        ...                                              amp=20)}
+        >>> sim = p2p.Simulation("ArgusI example", implant)
+        >>> resp = sim.pulse2percept(stim, implant)  # doctest: +SKIP
+        """
+        self._init()
+
+        # Parse `stim` (either single pulse train or a list/dict of pulse
+        # trains), and generate a list of pulse trains, one for each electrode
+        pt_list = p2p.stimuli.parse_pulse_trains(stim, self.implant)
+
+        self.gcl.tsample = pt_list[0].tsample
+
+        # Perform any necessary calculations per electrode
+        pt_list = p2p.utils.parfor(self.gcl.calc_per_electrode,
+                                   pt_list, engine=self.engine,
+                                   n_jobs=self.num_jobs)
+
+        # Which layer to simulate is given by implant type.
+        # For now, both implant types process the same two layers. In the
+        # future, these layers might differ. Order doesn't matter.
+        if self.implant.etype == 'epiretinal':

```

Comment:
Add an input for `layers` which will allow you to bypass these altogether. That way, you can input (for example): `layers = ['NFL']` (which you currently can't).  

Additionally, add a 'GCL' layer, for the ganglion cells. That way, our first set of simulations will pertain to a model that has `layers=['NFL', 'GCL']` (which is neither of the currently existing options). 

This also allows us to leave the current `epiretinal` and `subretinal` options and keep their functionality